CIPs/cip-0032.md

```diff
@@ -0,0 +1,117 @@
+
+1. The validator has been elected continuously for the past `B` blocks.
+2. The issuer was selected for at least one attestation in the past `B` blocks.
+3. The validator wasn't slashed by the `AttestationSlasher` within the past `B` blocks.
+4. The address hasn't completed an attestation in the past `B` blocks.
+5. The addres hasn't submitted `SignalOnline` event in the last `S` blocks. (see [below](#signal-online)).
+
+If the validator was intermittently elected, the contract will only consider the past `B` blocks in which the validator was *continuously* elected. For example, if a validator with a downed attestation node was elected for .5 `B` blocks, was unelected, and re-elected for .5 `B` blocks, then it will not be eligible for slashing since the slasher restarts counting when the validator was re-elected. Additionally, if the validator was elected but was never selected to provide an attestation in the entire period of `B` blocks, then they are not eligible for slashing.
+
+### Signal online
+
+In the unlikely event that a validator was never selected for an attestation during the period of `B` blocks, it's possible for a malicious actor to request multiple attestations with the goal of randomly getting paired with this validator and intentionally leaving the attestation incomplete. To prevent this, we're building into the AS the ability to signal to the `AttestationSlasher` that the service is still online if a slashing event becomes close by signing and submitting a transaction to the slasher contract. This requires that the AS signer key to be minimally funded to pay the gas fees of the transaction, however this adds more piece of mind for validators to be sure they won't be incorrectly slashed. Additionally, we'll be adding two new Prometheus events to the AS to alert a validator when the slashing period is nearly reached and another event to alert when the AS signer key's balance is low.
+
+If we observe that a validator is relying too heavily upon the `SignalOnline` functionality when they're receiving legitimate attestation requests we can add mitigations such as introducing an increasing financial penalty for subsquent signals or by decreasing the time that signal will last for (lowering `S`) per user.
+
+### Requiring Attestation Service
+
+**Note that the slasher expects each validator to be running an Attestation Service.** Unlike today, where it's optional for a Validator to run an attestation node, this proposal would make it mandatory. Having more attestation nodes in the pool improves the security of the phone number verification by making it more difficult to game the random selection of a set of colluding issuers. This proposal recommends providing at least three weeks from time of merge of this CIP to the time of enacting the slashing mechanism to give everyone an opportunity to set up their Attestation Service.
```

Comment:
I like the spirit of diversifying the attestors, but also requiring reluctant validators to run AS might decrease the overall reliability of the attestation protocol

Reply:
I hear the concern. Currently, 96/100 elected validators have registered Attestation Service claims so I don't think it will be too big of a lift. The alternative is that we allow validators to opt-out by deregistering an AS, but this doesn't feel fair to those that opt-in to support the network.